internal/source/git.go

```diff
@@ -259,38 +272,15 @@ func getProxyOpts(ctx context.Context, c client.Client, repo *sourcev1.GitReposi
 	if repo.Spec.ProxySecretRef == nil {
 		return nil, nil, nil
 	}
-	name := repo.Spec.ProxySecretRef.Name
-	namespace := repo.GetNamespace()
-	proxyData, err := getSecretData(ctx, c, name, namespace)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to get proxy secret '%s/%s': %w", namespace, name, err)
-	}
-	b, ok := proxyData["address"]
-	if !ok {
-		return nil, nil, fmt.Errorf("invalid proxy secret '%s/%s': key 'address' is missing", namespace, name)
-	}
-
-	address := string(b)
-	username := string(proxyData["username"])
-	password := string(proxyData["password"])
-
-	proxyOpts := &transport.ProxyOptions{
-		URL:      address,
-		Username: username,
-		Password: password,
+	secretRef := types.NamespacedName{
+		Name:      repo.Spec.ProxySecretRef.Name,
+		Namespace: repo.GetNamespace(),
 	}
-
-	proxyURL, err := url.Parse(string(address))
+	proxyURL, err := secrets.ProxyURLFromSecretRef(ctx, c, secretRef)
 	if err != nil {
-		return nil, nil, fmt.Errorf("invalid address in proxy secret '%s/%s': %w", namespace, name, err)
+		return nil, nil, fmt.Errorf("failed to get proxy URL from secret '%s/%s': %w", secretRef.Namespace, secretRef.Name, err)
```

Comment:
The `ProxyURLFromSecretRef` already has this reference to the secret: https://github.com/fluxcd/pkg/blob/6bf77f094c735b47a408cd1253c93939f33e1fed/runtime/secrets/converter.go#L152-L158